api/buckets/buckets.go

```diff
@@ -22,20 +37,56 @@ var (
 )
 
 type Bucket struct {
-	Key       string `json:"_id"`
-	Name      string `json:"name"`
-	Path      string `json:"path"`
-	DNSRecord string `json:"dns_record,omitempty"`
-	CreatedAt int64  `json:"created_at"`
-	UpdatedAt int64  `json:"updated_at"`
+	Key       string   `json:"_id"`
+	Name      string   `json:"name"`
+	Path      string   `json:"path"`
+	DNSRecord string   `json:"dns_record,omitempty"`
+	Archives  Archives `json:"archives"`
+	CreatedAt int64    `json:"created_at"`
+	UpdatedAt int64    `json:"updated_at"`
+}
+
+type Archives struct {
+	Current Archive   `json:"current"`
+	History []Archive `json:"history"`
+}
+
+type Archive struct {
+	Cid   string `json:"cid"`
+	Deals []Deal `json:"deals"`
+}
+
+type Deal struct {
+	ProposalCid string `json:"proposalCid"`
```

Comment:
Nit: Camelcase to match others?

Reply:
You mean:
```
ProposalCID string `json:"proposalCid"`
```
Or:
```
ProposalCid string `json:"proposal_cid"`
```
Or some other combination? :P

Reply:
Oop, the second one!

Reply:
Sure, will change.